recipes/pcapplusplus/all/conanfile.py

```diff
@@ -0,0 +1,97 @@
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class PcapplusplusConan(ConanFile):
+    name = "pcapplusplus"
+    version = "21.05"
+    license = "Unlicense"
+    description = "PcapPlusPlus is a multiplatform C++ library for capturing, parsing and crafting of network packets"
+    topics = ("conan", "pcapplusplus", "pcap", "network", "security", "packet")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/seladb/PcapPlusPlus"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "fPIC": [True, False],
+        "immediate_mode": [True, False],
+    }
+    default_options = {
+        "fPIC": True,
+        "immediate_mode": False,
+    }
+
+    exports_sources = "patches/*"
+    generators = "make"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.settings.os == "Windows":
+            # FIXME: missing winpcap recipe (https://github.com/bincrafters/community/pull/1395)
+            raise ConanInvalidConfiguration("winpcap is not available on cci (yet)")
+        else:
+            self.requires("libpcap/1.9.1")
+
+    def validate(self):
+        if self.settings.os not in ("FreeBSD", "Linux", "Macos"):
+            raise ConanInvalidConfiguration("%s is not supported" % self.settings.os)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+            destination=self._source_subfolder, strip_root=True)
+
+    @property
+    def _configure_sh_script(self):
+        return {
+            "Android": "configure-android.sh",
+            "Linux": "configure-linux.sh",
+            "Macos": "configure-mac_os_x.sh",
+            "FreeBSD": "configure-freebsd.sh",
+        }[str(self.settings.os)]
+
+    def _patch_sources(self):
+        if not self.options.get_safe("fPIC"):
+            tools.replace_in_file(os.path.join(self._source_subfolder, "PcapPlusPlus.mk.common"),
+                                  "-fPIC", "")
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def build(self):
+        self._patch_sources()
+        with tools.chdir(self._source_subfolder):
+            config_args = [
+                "./{}".format(self._configure_sh_script),
+                "--libpcap-include-dir", tools.unix_path(self.deps_cpp_info["libpcap"].include_paths[0]),
+                "--libpcap-lib-dir", tools.unix_path(self.deps_cpp_info["libpcap"].lib_paths[0]),
+            ]
+            if self.options.immediate_mode:
+                config_args.append("--use-immediate-mode")
+            if tools.is_apple_os(self.settings.os) and "arm" in self.settings.arch:
```

Comment:
Can the arm arch also be 32-bit? eg armv7

Reply:
I don't think Apple M1 supports 32-bit 🤔

Reply:
So testing for armv8 is not needed, I guess?
I believe armv9 is in the pipeline so let's keep it like this now.